### PR TITLE
feat: initial commit

### PR DESCRIPTION
**Pull request by metcoder95**
Initial PR for cloud-pine

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1 +1,16 @@
...
+const split = require('split2')
+const { CloudLogging } = require('./lib/cloud-logging')
+
+async function CloudPine (options) {
+  const { logName = 'Cloud_Pine', cloudLoggingOptions } = options
+  const logging = new CloudLogging(logName, cloudLoggingOptions)
+
+  await logging.init()
+
+  const stream = split(logging.parseLine.bind(logging), { autoDestroy: true })
```
Comment (kruczyna) [inline index.js:11]: I am a bit confused here:
```
logging.parseLine.bind(logging),
```
Why do we borrow and assign `logging` to `logging`? 🤔

**Review by kruczyna — APPROVED** [review]
LGTM 👍  
I'm just fuzzy in one moment - left a comment 😄

File: index.js (hunk 1 above)
Comment (metcoder95) [reply index.js:11]: It's just to inject the function without losing the context of the `this`.
If I just borrow the function, the `parseLine` uses the `this` to access a couple of helper functions and the `Logging` instance, all of the previous will be lost if I don't properly hard-bind the `logging` instance to the `bind` function itself

**Comment by metcoder95** [comment]
Let's merge and continue with adding testing in a next PR 👍

**Pull request merged**